influxql/ast.go

```diff
@@ -954,11 +954,14 @@ func (s *SelectStatement) ColumnNames() []string {
 		case *Call:
 			if f.Name == "top" || f.Name == "bottom" {
 				if len(f.Args) == 2 {
-					columnNames = append(columnNames, f.Name)
+					columnNames = append(columnNames, field.Name())
 					continue
 				}
 				// We have a special case now where we have to add the column names for the fields TOP or BOTTOM asked for as well
-				columnNames = slices.Union(columnNames, f.Fields(), true)
```

Comment:
This union function was always used a bit weirdly. I think it would be better to just modify the function to be something like this:

```
columnNames = append(columnNames, field.Name())
if len(f.Args) > 2 {
    names := make([]string, 0, len(f.Args)-2)
    for i := 1; i < len(f.Args) - 1; i++ {
        ref := f.Args[i].(*VarRef)
        columnNames = append(columnNames, ref.Val)
    }
}
```

It's more simple than using union and uses the same code path for adding the first column in both circumstances. Union also isn't really the correct operation as this is supposed to add a column regardless of whether it already exists in the list of column names, but I kept it because the previous code used it and I didn't want to change things too much.